### PR TITLE
Feature: Refactor OWASP dependency check runs

### PR DESCRIPTION
**Pull request by infeo**
This PRs changes the way, how we use the https://github.com/jeremylong/DependencyCheck maven plugin.

Changes were necessary due to the deprecation of an API used by dependency check (see [here](https://github.com/jeremylong/DependencyCheck#900-upgrade-notice) for details).

Changes:
* Plugin updated to 9.0.4
* plugin uses an NVD API key
* only before a release the plugin is executed
* once a week, the plugin is run on CI. If a vulnerability is detected, the workflow fails and the report is uploaded.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -40,6 +40,11 @@ jobs:
         with:
           name: artifacts
           path: target/*.jar
+      - name: Check dependencies for releases
+        if: startsWith(github.ref, 'refs/tags/')
```
Comment (overheadhunter) [inline .github/workflows/build.yml:44]: Don't run on tagged commits, but rather on release-branches. Failing here might be too late, as in some repos a lot of automation happens when pushing tags.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ updates:
       interval: "weekly"
       day: "monday"
       time: "06:00"
-      timezone: "UTC"
+      timezone: "Etc/UTC"
```
Comment (overheadhunter) [inline .github/dependabot.yml:9]: unrelated

File: .github/workflows/dependency-check.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          show-progress: false
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: 'maven'
+      - name: Run org.owasp:dependency-check plugin
+        id: plugin-run
```
Comment (overheadhunter) [inline .github/workflows/dependency-check.yml:23]: choose a more descriptive name, e.g.

```suggestion
        id: dependency-check
```

**Review by overheadhunter — CHANGES_REQUESTED** [review]
good start, but how are we notified? just a failed workflow? that might be a lot of spam for the repo owner. instead, can we post such notifications in slack?

**Comment by infeo** [comment]
> good start, but how are we notified? just a failed workflow? that might be a lot of spam for the repo owner. instead, can we post such notifications in slack?

I had exactly the same thoughts with the reversed result 😆  But will change it to a slack notification.

File: .github/workflows/dependency-check.yml (hunk 3 above)
Comment (infeo) [reply .github/workflows/dependency-check.yml]: Changed in https://github.com/cryptomator/cryptofs/pull/192/commits/60b98e43a14a8b3d7e2bb835307cbef05c4ecf4b

File: .github/dependabot.yml (hunk 2 above)
Comment (infeo) [reply .github/dependabot.yml]: Reverted in https://github.com/cryptomator/cryptofs/pull/192/commits/60b98e43a14a8b3d7e2bb835307cbef05c4ecf4b

File: .github/workflows/build.yml (hunk 1 above)
Comment (infeo) [reply .github/workflows/build.yml]: Changed in https://github.com/cryptomator/cryptofs/pull/192/commits/60b98e43a14a8b3d7e2bb835307cbef05c4ecf4b

**Comment by infeo** [comment]
> notifications in slack

Done in https://github.com/cryptomator/cryptofs/pull/192/commits/8e20b707f52e62dd33db206ed6bf827c9c06df15.

@overheadhunter The only part missing is now the merge constraint. Can you add it to the repository? It should be very similar to https://github.com/cryptomator/cryptomator and its `release-check.yml` action.

File: .github/workflows/dependency-check.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          NVD_API_KEY: ${{ secrets.NVD_API_KEY }}
+      - name: Upload report on failure
+        if: "${{ steps.dependency-check.outcome == 'failure'}}"
+        uses: actions/upload-artifact@v3
+        with:
+          name: dependency-check-report
+          path: target/dependency-check-report.html
+          if-no-files-found: error
+      - name: Slack Notification on regular check
+        if: github.event_name == 'schedule'
```
Comment (overheadhunter) [inline .github/workflows/dependency-check.yml:40]: Doesn't this need some `&& foo == failure`? Or does this only apply to the first consecutive step after running the depencency check?

**Comment by overheadhunter** [comment]
> The only part missing is now the merge constraint. Can you add it to the repository?

I can't reference the action from the repo settings, before it exists on the default branch

File: .github/workflows/dependency-check.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          java-version: 17
+          cache: 'maven'
+      - name: Run org.owasp:dependency-check plugin
+        id: dependency-check
+        continue-on-error: true
+        run: mvn -B verify -Pdependency-check -DskipTests
+        env:
+          NVD_API_KEY: ${{ secrets.NVD_API_KEY }}
+      - name: Upload report on failure
+        if: "${{ steps.dependency-check.outcome == 'failure'}}"
```
Comment (overheadhunter) [inline .github/workflows/dependency-check.yml:32]: can't this be just

```suggestion
        if: steps.dependency-check.outcome == 'failure'
```

**Review by overheadhunter — CHANGES_REQUESTED** [review]

**Review by overheadhunter — APPROVED** [review]

File: .github/workflows/dependency-check.yml (hunk 5 above)
Comment (infeo) [reply .github/workflows/dependency-check.yml]: Done in https://github.com/cryptomator/cryptofs/pull/192/commits/d523d09817feef33e03e86c58a0c5a696481543b
